tests/device_tests/test_shamir_reset_recovery_groups.py

```diff
@@ -28,22 +28,26 @@
 def test_reset_recovery(client):
     mnemonics = reset(client)
     address_before = btc.get_address(client, "Bitcoin", parse_path("44'/0'/0'/0/0"))
-    # TODO: more combinations
-    selected_mnemonics = [
-        mnemonics[0],
-        mnemonics[1],
-        mnemonics[2],
-        mnemonics[5],
-        mnemonics[6],
-        mnemonics[7],
-        mnemonics[10],
-        mnemonics[11],
-        mnemonics[12],
-    ]
-    device.wipe(client)
-    recover(client, selected_mnemonics)
-    address_after = btc.get_address(client, "Bitcoin", parse_path("44'/0'/0'/0/0"))
-    assert address_before == address_after
+    # we're generating 3of5 groups 3of5 shares each
+    for group_offset in range(3):
```

Comment:
Is there a better way to do these combinations?

Reply:
something like this?

```python
groups = [mnemonics[i*5 : i*5 + 5] for i in range(5)]
for selected_groups in itertools.combinations(groups, 3):
    for indices in itertools.combinations(range(5), 3):
         selected_mnemonics = [
             group[idx]
             for group in selected_groups
             for idx in indices
         ]
         # ...
```

it does not test *every* possible combination, but it's 100 variants as it is

Reply:
thanks for the suggestion. I'll have a look at it. We don't need to test all the combinations, just all the shares/groups. The number of test combinations can be reduced since currently we're testing some of the shares and groups multiple times. I'll try to do that too.

Reply:
+1 for splitting up the mnemonics to groups first to make it a bit more readable

Reply:
In the end I decided to create the test combinations manually. To test all shares from all groups we need 4 combinations. Trying various combinations between shares/groups we already tested is probably an overkill.